pkg/services/sqlstore/annotation.go

```diff
@@ -223,7 +225,7 @@ func (r *SqlAnnotationRepo) Find(query *annotations.ItemQuery) ([]*annotations.I
 		query.Limit = 100
 	}
 
-	sql.WriteString(" ORDER BY epoch DESC" + dialect.Limit(query.Limit))
+	sql.WriteString(" ORDER BY a.org_id, a.epoch_end DESC, a.epoch DESC" + dialect.Limit(query.Limit) + " ) dt on dt.id = annotation.id")
```

Comment:
```suggestion
	// order of ORDER BY arguments match the order of a sql index for performance 
	sql.WriteString(" ORDER BY a.org_id, a.epoch_end DESC, a.epoch DESC" + dialect.Limit(query.Limit) + " ) dt on dt.id = annotation.id")
```

Reply:
@marefr what is the performance for other queries where another index is picked instead of `IDX_annotation_org_id_epoch_end_epoch`
maybe we need to change the ordering by according to the index (the where condition)

Reply:
Using `IDX_annotation_org_id_dashboard_id_epoch_end_epoch`

```
PANEL ALERT TAB:
SELECT   annotation.id,   annotation.epoch as time,   annotation.epoch_end as time_end,   annotation.dashboard_id,   annotation.panel_id,   annotation.new_state,   annotation.prev_state,   annotation.alert_id,   annotation.text,   annotation.tags,   annotation.data,   annotation.created,   annotation.updated,   usr.email,   usr.login,   alert.name as alert_name FROM annotation LEFT OUTER JOIN user as usr on usr.id = annotation.user_id LEFT OUTER JOIN alert on alert.id = annotation.alert_id INNER JOIN (   SELECT a.id from annotation a     WHERE a.org_id = 1     AND a.dashboard_id = 78 and a.panel_id=2     AND a.alert_id > 0     ORDER BY a.org_id, a.epoch_end DESC, a.epoch DESC LIMIT 50 ) dt  ON dt.id = annotation.id;
50 rows in set (0.00 sec)

explain SELECT   annotation.id,   annotation.epoch as time,   annotation.epoch_end as time_end,   annotation.dashboard_id,   annotation.panel_id,   annotation.new_state,   annotation.prev_state,   annotation.alert_id,   annotation.text,   annotation.tags,   annotation.data,   annotation.created,   annotation.updated,   usr.email,   usr.login,   alert.name as alert_name FROM annotation LEFT OUTER JOIN user as usr on usr.id = annotation.user_id LEFT OUTER JOIN alert on alert.id = annotation.alert_id INNE
+----+-------------+------------+--------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------+---------+-----------------------------+-------+----------------------------------------------------+
| id | select_type | table      | type   | possible_keys                                                                                                                                                                                                  | key                                                | key_len | ref                         | rows  | Extra                                              |
+----+-------------+------------+--------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------+---------+-----------------------------+-------+----------------------------------------------------+
|  1 | PRIMARY     | <derived2> | ALL    | NULL                                                                                                                                                                                                           | NULL                                               | NULL    | NULL                        |    50 | NULL                                               |
|  1 | PRIMARY     | annotation | eq_ref | PRIMARY                                                                                                                                                                                                        | PRIMARY                                            | 8       | dt.id                       |     1 | NULL                                               |
|  1 | PRIMARY     | usr        | ALL    | PRIMARY                                                                                                                                                                                                        | NULL                                               | NULL    | NULL                        |     1 | Using where; Using join buffer (Block Nested Loop) |
|  1 | PRIMARY     | alert      | eq_ref | PRIMARY                                                                                                                                                                                                        | PRIMARY                                            | 8       | grafana.annotation.alert_id |     1 | NULL                                               |
|  2 | DERIVED     | a          | ref    | IDX_annotation_org_id_alert_id,IDX_annotation_org_id_type,IDX_annotation_org_id_created,IDX_annotation_org_id_updated,IDX_annotation_org_id_dashboard_id_epoch_end_epoch,IDX_annotation_org_id_epoch_end_epoch | IDX_annotation_org_id_dashboard_id_epoch_end_epoch | 17      | const,const                 | 29068 | Using where                                        |
+----+-------------+------------+--------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------+---------+-----------------------------+-------+----------------------------------------------------+
```